src/chains/filecoin/src/api.ts

```diff
@@ -0,0 +1,106 @@
+//#region Imports
+import { types } from "@ganache/utils";
+import Blockchain from "./blockchain";
+import {
+  StorageProposal,
+  SerializedStorageProposal
+} from "./things/storageproposal";
+import { SerializedRootCID, RootCID } from "./things/rootcid";
+import { SerializedDeal } from "./things/deal";
+import { SerializedTipset } from "./things/tipset";
+import { SerializedAddress } from "./things/address";
+import { SerializedMiner } from "./things/miner";
+import {
+  SerializedRetrievalOffer,
+  RetrievalOffer
+} from "./things/retrievaloffer";
+
+const _blockchain = Symbol("blockchain");
+
+export default class FilecoinApi implements types.Api {
+  readonly [index: string]: (...args: any) => Promise<any>;
+
+  private readonly [_blockchain]: Blockchain;
+
+  constructor(blockchain: Blockchain) {
+    this[_blockchain] = blockchain;
+  }
+
+  async stop(): Promise<void> {
+    return await this[_blockchain].stop();
+  }
+
+  async "Filecoin.ChainGetGenesis"(): Promise<SerializedTipset> {
```

Comment:
@davidmurdoch @tcoulter why do these function names need periods, requiring them to be declared with string literals? Why not just have `FilecoinChainGetGenesis` or `Filecoin_ChainGetGenesis`?

Reply:
Thanks @tcoulter for the [clarification](https://github.com/trufflesuite/ganache-core/pull/681#issuecomment-749111969)!